pkg/api/handlers/projectmcp.go

```diff
@@ -0,0 +1,594 @@
+	}
+
+	resources, err := p.mcpSessionManager.ListResources(req.Context(), req.User.GetUID(), server, serverConfig)
+	if err != nil {
+		var are nmcp.AuthRequiredErr
+		if strings.HasSuffix(err.Error(), "Method not found") {
+			return types.NewErrHTTP(http.StatusFailedDependency, "MCP server does not support resources")
+		} else if errors.As(err, &are) {
+			return types.NewErrHTTP(http.StatusPreconditionFailed, "MCP server requires authentication")
+		}
+		return fmt.Errorf("failed to list resources: %w", err)
+	}
+
+	return req.Write(resources)
+}
+
+func (p *ProjectMCPHandler) ReadResource(req api.Context) error {
+	var projectServer v1.ProjectMCPServer
+	err := req.Get(&projectServer, req.PathValue("project_mcp_server_id"))
+	if err != nil {
+		return err
+	}
+
+	var (
+		server       v1.MCPServer
+		serverConfig mcp.ServerConfig
+	)
+	if system.IsMCPServerInstanceID(projectServer.Spec.Manifest.MCPID) {
+		server, serverConfig, err = ServerFromMCPServerInstance(req, projectServer.Spec.Manifest.MCPID)
+	} else {
+		server, serverConfig, err = ServerForActionWithID(req, projectServer.Spec.Manifest.MCPID)
+	}
+	if err != nil {
+		return err
+	}
+
+	caps, err := p.mcpSessionManager.ServerCapabilities(req.Context(), req.User.GetUID(), server, serverConfig)
+	if err != nil {
+		return err
+	}
+
+	if caps.Resources == nil {
+		return types.NewErrHTTP(http.StatusFailedDependency, "MCP server does not support resources")
+	}
+
+	contents, err := p.mcpSessionManager.ReadResource(req.Context(), req.User.GetUID(), server, serverConfig, req.PathValue("resource_uri"))
+	if err != nil {
+		var are nmcp.AuthRequiredErr
+		if strings.HasSuffix(err.Error(), "Method not found") {
+			return types.NewErrHTTP(http.StatusFailedDependency, "MCP server does not support resources")
+		} else if errors.As(err, &are) {
+			return types.NewErrHTTP(http.StatusPreconditionFailed, "MCP server requires authentication")
+		}
+		return fmt.Errorf("failed to list resources: %w", err)
+	}
+
+	return req.Write(contents)
+}
+
+func (p *ProjectMCPHandler) GetPrompts(req api.Context) error {
+	var projectServer v1.ProjectMCPServer
+	err := req.Get(&projectServer, req.PathValue("project_mcp_server_id"))
+	if err != nil {
+		return err
+	}
+
+	var (
+		server       v1.MCPServer
+		serverConfig mcp.ServerConfig
+	)
+	if system.IsMCPServerInstanceID(projectServer.Spec.Manifest.MCPID) {
+		server, serverConfig, err = ServerFromMCPServerInstance(req, projectServer.Spec.Manifest.MCPID)
+	} else {
+		server, serverConfig, err = ServerForActionWithID(req, projectServer.Spec.Manifest.MCPID)
+	}
+	if err != nil {
+		return err
+	}
+
+	caps, err := p.mcpSessionManager.ServerCapabilities(req.Context(), req.User.GetUID(), server, serverConfig)
+	if err != nil {
+		return err
+	}
+
+	if caps.Resources == nil {
+		return types.NewErrHTTP(http.StatusFailedDependency, "MCP server does not support resources")
+	}
```

Comment:
Why do we check if resources are supported here? Does the prompt capability also require the resource capability?

Reply:
No, AI cruft that I should have caught.